CIP-9999/README.md

```diff
@@ -0,0 +1,189 @@
+
+<details>
+  <summary><span id="header-preamble"><strong>Header preamble</strong></span></summary>
+
+Each CIP must begin with a YAML key:value style header preamble (also known as 'front matter data'), preceded and followed by three hyphens (`---`).
+
+Field                | Description
+---                  | ---
+`CPS`                | CPS number (without leading 0), or "\?" before being assigned
+`Title`              | A succinct and descriptive title
+`Status`             | Open \| Solved \| Inactive (..._reason_...)
+`Category`           | One registered or well-known category covering one area of the ecosystem.
+`Authors`            | A list of authors' real names and email addresses (e.g. John Doe <john.doe@email.domain>)
+`Proposed Solutions` | A list of CIPs addressing the problem, if any
+`Discussions`        | A list of links where major technical discussions regarding this CPS happened. Links should include any discussion before submission, a link to the pull request that created the CPS, and any pull request that modifies it.
+`Created`            | Date created on, in ISO 8601 (YYYY-MM-DD) format
+
+For example:
+
+```yaml
+---
+CPS: 1
+Title: The Blockchain Trilemma
+Status: Open
+Category: Consensus
+Authors:
+    - Alice <alice@domain.org>
+    - Bob <bob@domain.org>
+Proposed Solutions: []
+Discussions:
+    - https://forum.cardano.org/t/solving-the-blockchain-trilemma/107720
+    - https://github.com/cardano-foundation/cips/pulls/9999
+Created: 2009-02-14
+---
+```
+</details>
+
+<details>
+  <summary><span id="repository-organization"><strong>Repository Organization</strong><span></summary>
+
+A CPS must be stored in a specific folder named after its number and in a file called `README.md`. Before a number is assigned, use `????` as a placeholder name.
+
+Additional supporting files (such as diagrams, binary specifications, dialect grammars, JSON schemas etc.) may be added to the CPS's folder under freely chosen names.
+
+For example:
+
+```
+CPS-0001
+├── README.md
+└── requirements.toml
+```
+</details>
+
+> **Note** A reference template is available in [.github/CPS-TEMPLATE.md][CPS-TEMPLATE.md]
+
+#### Statuses
+
+From its creation onwards, a problem statement evolves around the following statuses.
+
+Status       | Description
+---          | ---
+Open         | Any problem statement that is fully formulated but for which there still exists no solution that meets its goals. Problems that are only partially solved shall remain _'Open'_ and list proposed solutions so far in their header's preamble.
+Solved       | Problems for which one or more solutions have been found and implemented. When solved via a CIP, the solved status should indicate it as such: `Solved: by <CIP-XXXX>[,<CIP-YYYY>,...]`.
```

Comment:
Given my previous comment about CPSs probably having multiple CIPs that "solve" parts of them, I wonder again about having something like "Addressed-by" instead. And perhaps the CPS could move to "Resolved" when it is completely addressed? Because that could also happen if it ceases to be a problem for some other reason.

Reply:
There already is a `Proposed Solutions` field in the preamble. Having another `Addressed-By` field feels redundant since, any `Proposed Solutions` in an `Active` would be effectively addressing (even partially) the problem. If you're suggesting to rename `Proposed Solutions` to `Addressed-By`, that's an option. I have no strong feeling about naming here.

Reply:
I guess it just seems weird to me to put the CPS in the Solved state if only part of the problem has been addressed. That seems misleading. But then, leaving it in Open when it has been partially solved is perhaps also misleading. So I don't know what to do.

Reply:
Then I need to clarify something because, `Solved` is only assigned to problems that are **fully** solved. A problem that is only partially solved remains in `Open`, but is updated accordingly to show what's left to solve. 

As @abailly-iohk also hinted once, it's probably better to have CPS be focused on a single "atomic" problem as much as possible to avoid this type of situation.

Reply:
Well, the text says just:

> Problems for which one or more solutions have been found and implemented. 

Which certainly has left me confused about the partial solution state.

> As @abailly-iohk also hinted once, it's probably better to have CPS be focused on a single "atomic" problem as much as possible to avoid this type of situation.

I don't think I agree with this. I think it's very unlikely that the "problem view" lines up with the "solution view" so neatly. In practice we often need to do many things to fully address a user-visible problem!

Reply:
The important part is 
> as much as possible
Of course, real life is messy and problems do not usually align neatly with solutions, but we should encourage definitions of problems that are actionable and "compact", eg. instead of a CPS trying to "Solve world hunger", write a CPS saying "How to salvage France's hypermarkets food trashed?" 
Not sure this is a great example but I guess the idea is there

Reply:
> "How to salvage France's hypermarkets food trashed?"

... which is hardly going to have one solution, no? :) I just think it's unrealistic to suppose that every CPS will have one solution. In fact, my bet would be that that's the less common case.

Reply:
That a problem stated in a CPS has more than one solution is certainly desirable and that's actually what I was, awkwardly, trying to express.

Reply:
The idea I was trying to convey is that a CPS should express a problem that has  _at least_ one solution which itself could be expressed as a CIP. We want to rule out problems whose solution required something that would not fit in a single CIP (I think).